external_deps/build.sh

```diff
@@ -1159,17 +1232,74 @@ if [ "${#}" -lt "2" ]; then
 
 	EOF
 	false
+}
+
+download_only='false'
+prefer_ours='false'
+while [ -n "${1:-}" ]
+do
+	case "${1-}" in
+	'--download-only')
+		download_only='true'
+		shift
+	;;
+	'--prefer-ours')
+		prefer_ours='true'
+		shift
+	;;
+	'--'*)
+		helpError
+	;;
+	*)
+		break
+	esac
+done
+
+# Usage
+if [ "${#}" -lt "2" ]; then
+	errorHelp
 fi
 
 # Enable parallel build
 export MAKEFLAGS="-j`nproc 2> /dev/null || sysctl -n hw.ncpu 2> /dev/null || echo 1`"
 
 # Setup platform
-PLATFORM="${1}"; shift
-"setup_${PLATFORM}"
-
-# Build packages
-for pkg in "${@}"; do
-	cd "${WORK_DIR}"
-	"build_${pkg}"
+platform="${1}"; shift
+
+platform_list=''
+case "${platform}" in
+'all')
+	platform_list="${all_platforms}"
+;;
+'build-linux')
+	platform_list="${linux_build_platforms}"
+;;
+'build-macos')
+	platform_list="${macos_build_platforms}"
+;;
+*)
+	for known_platform in ${all_platforms}
+	do
+		if [ "${platform}" = "${known_platform}" ]
+		then
+			platform_list="${platform}"
+			break;
+		fi
+	done
+	if [ -z "${platform_list}" ]
+	then
+		errorHelp
+	fi
+;;
+esac
+
+for PLATFORM in ${platform_list}
+do
+	"setup_${PLATFORM}"
```

Comment:
I have doubts that it is really safe to run more than one setup platform. Some of them might set environment variables that should not be set for others. I suppose it could be worked around by running each platform in a new process.

Reply:
Ah, yes, I just have to add some `()`

Reply:
Fixed.